presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -547,10 +548,11 @@ private static boolean isDirectory(HdfsContext context, HdfsEnvironment hdfsEnvi
         }
     }
 
-    public static Path createTemporaryPath(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path targetPath)
+    public static Path createTemporaryPath(ConnectorSession session, HdfsContext context, HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        String temporaryPrefix = "/tmp/presto-" + context.getIdentity().getUser();
+        String temporaryPrefix = getTemporaryStagingDirectoryLocation(session)
```

Comment:
I am curious about how `temporaryRoot` get formed on line 563.

Does that mean if the `targetPath` is `hdfs://path/to/your/table`, then `temporaryRoot` will be `hdfs://path/to/your/table/tmp/presto-userid/` ? I thought temporary path should be something like `hdfs://tmp/presto-userid` ?...

Reply:
never mind. the path constructor doesn't construct sub directory.